packages/deployer/src/compiler.ts

```diff
@@ -254,4 +304,25 @@ export class Compiler {
             await fsWrapper.mkdirAsync(this._artifactsDir);
         }
     }
+    /**
+     * Gets contract data on network or returns if an artifact does not exist.
+     * @param fileName Name of contract file.
+     * @return Contract data on network or undefined.
+     */
+    private async _getContractArtifactIfExistsAsync(fileName: string): Promise<ContractArtifact | void> {
+        let contractArtifact;
+        const contractName = path.basename(fileName, constants.SOLIDITY_FILE_EXTENSION);
+        const currentArtifactPath = `${this._artifactsDir}/${contractName}.json`;
+        try {
+            const opts = {
+                encoding: 'utf8',
+            };
+            const contractArtifactString = await fsWrapper.readFileAsync(currentArtifactPath, opts);
+            contractArtifact = JSON.parse(contractArtifactString);
+            return contractArtifact;
+        } catch (err) {
+            utils.consoleLog(`Artifact for ${fileName} does not exist`);
+            return contractArtifact;
```

Comment:
Let's just return `undefined` explicitly here so that it's clearer to the reader what is happening. This confused me for a second and could be brittle to future changes.